Sources/Sentry/SentryTracer.m

```diff
@@ -199,11 +216,135 @@ - (SentryTransaction *)toTransaction
                                             }]];
     }
 
+    if (appStartMeasurement != nil) {
+        [self setStartTimestamp:appStartMeasurement.appStartTimestamp];
+    }
+
     SentryTransaction *transaction = [[SentryTransaction alloc] initWithTrace:self children:spans];
     transaction.transaction = self.name;
+    [self addMeasurements:transaction appStartMeasurement:appStartMeasurement];
     return transaction;
 }
 
+- (SentryAppStartMeasurement *)getAppStartMeasurment
+{
+    SentryAppStartMeasurement *appStartMeasurement = nil;
+
+    // Only send app start measurement for transactions generated by auto performance
+    // instrumentation.
+    if (![self.context.operation isEqualToString:SENTRY_VIEWCONTROLLER_RENDERING_OPERATION]) {
+        return appStartMeasurement;
+    }
+
+    // Double-Checked Locking to avoid acquiring unnecessary locks.
+    if (SentrySDK.appStartMeasurement == nil) {
+        return appStartMeasurement;
+    }
+
+    @synchronized(SentrySDK.appStartMeasurementLock) {
+        if (SentrySDK.appStartMeasurement != nil) {
+            NSDate *appStartTimestamp = SentrySDK.appStartMeasurement.appStartTimestamp;
+            NSDate *appStartEndTimestamp =
+                [appStartTimestamp dateByAddingTimeInterval:SentrySDK.appStartMeasurement.duration];
+
+            NSTimeInterval difference =
+                [appStartEndTimestamp timeIntervalSinceDate:self.startTimestamp];
+
+            // If the difference between the end of the app start and the beginning of the current
+            // transaction is smaller than SENTRY_APP_START_MEASUREMENT_DIFFERENCE. With this we
+            // avoid messing up transactions too much.
+            if (difference <= SENTRY_APP_START_MEASUREMENT_DIFFERENCE
+                && difference >= -SENTRY_APP_START_MEASUREMENT_DIFFERENCE) {
+                appStartMeasurement = SentrySDK.appStartMeasurement;
```

Comment:
We'll read it then only once to make sure this only goes out to Sentry once, which makes sense. One thing I wanted to point out is that we'll want to pull this from React Native and other hybrids. So we need (future problem, future PR) but good to keep in mind that we'll need to have a way to expose this out for non `SENTRY_VIEWCONTROLLER_RENDERING_OPERATION`.

Reply:
I'm thankful that you look ahead. As this is internal and can be changed easily, I think this (famous last words 😬) shouldn't bite us in the future.